iroh-net/examples/listen.rs

```diff
@@ -62,9 +62,10 @@ async fn main() -> anyhow::Result<()> {
         "\tcargo run --example connect -- --node-id {me} --addrs \"{local_addrs}\" --relay-url {relay_url}\n"
     );
     // accept incoming connections, returns a normal QUIC connection
-    while let Some(conn) = endpoint.accept().await {
-        // accept the connection and extract the `node_id` and ALPN
-        let (node_id, alpn, conn) = iroh_net::magic_endpoint::accept_conn(conn).await?;
+    while let Some(mut conn) = endpoint.accept().await {
+        let alpn = conn.alpn().await?;
+        let conn = conn.await?;
+        let node_id = iroh_net::magic_endpoint::get_remote_node_id(&conn)?;
```

Comment:
why did this code get worse, can't we keep the helper?

Reply:
heh, you think this is worse?  To me the helper function was a long-standing eyesore.  This way we provide the same API as upstream Quinn does, with the `.alpn()` method extra.

If we did `.alpn()` as a helper function, you could write:
```rust
let conn = ep.accept().await?;
let node_id = get_remote_nodeid(&conn)?;
let alpn = get_alpn(&conn)?;
```

We could keep the helper function.  But then there's multiple ways to do something, which I'm not super keen on.  Though I guess if you know you need all three it's reasonable, but I've also seen code that did only need the connection and used the helper function because that's what they thought it should do.

Reply:
hmm I guess the nicest version would be to 

```rust
let alpn = conn.alpn().await?;
let conn = conn.await?;
let node_id = conn.remote_node_id()?;
```

Reply:
but overall, I agree, now that I look at it, this is actually better than before

Reply:
Once and if we wrap `Connection` we can even do:
```rust
let conn = ep.accept().await?;
// optionally, if you need either of these
let alpn = conn.alpn().await?;
let node_id = conn.remote_node_id()?;
```
which I think would be the nicest.  And you could still get the alpn from the intermediate `Connecting` future as well if you need it earlier.

Reply:
yes, agreed

Reply:
ok, re-exported `Connection` now.  Are we leave the rest of this API as now in the PR?